api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -33,6 +33,16 @@
  */
 public interface Context
 {
+  /*
```

Comment:
This is good thinking. A few comments:

1. The Note: This is a developer documentation - would like to see this as one of the unit tests instead of a comment here.

2.  There is already a disambiguation supported at the Java level by use of the context interface? The apex properties file support is disliked by users for various reasons. I personally did not like the lack of disambiguation when the AtrributeName is the same across various contexts. We need to allow properties file to enable disambiguation instead of allowing it by putting restrictions. so can properties file be enhanced to allow me to specify which attribute I am setting - DAGContext vs OperatorContext vs PortContext. 

3. The code treats DAGContext.ATTRIBUTE different than OperatorContext.ATTRIBUTE so this restriction does not help a bit to prevent the backward incompatibility. The correct solution would be getting 2 right.